h/auth/policy.py

```diff
@@ -42,6 +42,45 @@ def forget(self, request):
         return self.fallback_policy.forget(request)
 
 
+@interface.implementer(interfaces.IAuthenticationPolicy)
+class APIAuthenticationPolicy(object):
+    """
+    An authentication policy for Hypothesis API endpoints
+
+    Two types of authentication apply to Hypothesis API endpoints:
+
+    * Authentication for a single user using Token authentication
+    * Authentication for an auth_client, either as the client itself or
+      "on behalf of" a user within that client's authority
+
+    This policy delegates to :py:class:`~h.auth.TokenAuthenticationPolicy` and
+    :py:class:`~h.auth.AuthClientPolicy`, always preferring Token when available
+    """
+    def __init__(self, user_policy, client_policy):
+        self.user_policy = user_policy
+        self.client_policy = client_policy
```

Comment:
Might it be simpler if `APIAuthenticationPolicy` constructed `self.user_policy` and `self.client_policy` itself? Since I guess when this is hooked up `TokenAuthenticationPolicy`, `AuthClientPolicy` and `APIAuthenticationPolicy` are both only going to be instantiated one way, the flexibility of being able to pass in different `user_policy`'s and `client_policy`'s from the outside isn't really needed?

This might also help to clarify that `user_policy` is a `TokenAuthenticationPolicy` instance (the name changing from "token" to "user" muddies this a little).

I guess the tests could make use of the arguments to pass in mock `user_policy`'s and `client_policy`'s, but the tests can also just patch `TokenAuthenticationPolicy` and `AuthClientPolicy`.